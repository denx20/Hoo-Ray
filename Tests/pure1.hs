-- A function with only pure functions (i.e. only `let`s in main)
add :: Int -> Int -> Int -> Int
add x y z = x + y + z

sub :: Int -> Int -> Int
sub x y = x - y

multiply :: Int -> Int -> Int
multiply x y = x * y

divide :: Int -> Int -> Int
divide x y = x `div` y

main :: IO ()
main = do
    let x = 10
    let y = 5
    let z = 3
    let w = add x y z
    let k = multiply x z
    let a = k `sub` y
    print