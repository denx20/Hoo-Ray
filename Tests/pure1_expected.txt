("w", ["add"])
("add", ["x","y","z"])
("k", ["multiply"])
("multiply", ["x","z"])
("a", ["sub"])
("sub", ["k","y"])
